### ISSUE TITLE
The Remote API tasks are completely undocumented

### ISSUE THREAD

**Issue by chrisr3** [bmuschko/gradle-docker-plugin#737]
Trying to learn how to use the `docker-remote-api` Gradle plugin, and so I tried reading the [User Guide](https://bmuschko.github.io/gradle-docker-plugin/#remote_api_plugin). However, clicking through to the JavaDoc for the Gradle tasks revealed that they were completely blank (c.f. [DockerCreateContainer](https://bmuschko.github.io/gradle-docker-plugin/api/com/bmuschko/gradle/docker/tasks/container/DockerCreateContainer.html).)

Some co-workers sympathised with my predicament, revealing that they'd previously abandoned attempting to use this plugin and depended upon `docker-java-api` directly. They then advised me to do the same.

The revelation here is that the `docker-remote-api` plugin is actually wrapping Docker's REST API, and the `docker` CLI client does _not_ (necessarily).

The bottom line here is that the `docker-remote-api` plugin is woefully under-documented.

Seee #733.

**Comment by bmuschko** [comment]
Thanks for your feedback. We'll work on better documentation in the near future.

> Some co-workers sympathised with my predicament, revealing that they'd previously abandoned attempting to use this plugin and depended upon `docker-java-api` directly. They then advised me to do the same.

Many property values exposed by tasks of this plugin only delegate to the corresponding methods available in `docker-java`. I don't think the documentation would look much different. Either way, I think we could do a better job here.

> The revelation here is that the docker-remote-api plugin is actually wrapping Docker's REST API, and the docker CLI client does not (necessarily).

I am pretty sure we say that in the plugin's user guide e.g. "Gradle plugin for managing Docker images and containers using the Docker remote API. The heavy lifting of communicating with the Docker remote API is handled by the Docker Java library.". How do you think we could explain this more clearly so it becomes abundantly clear to other users?

**Comment by chrisr3** [comment]
My prior experience with Docker is with its CLI tool only - I had no idea that a RESTful API of JSON objects even _existed_. So that rather high-level statement didn't really register. Maybe emphasise that you are providing an alternative _low-level_ Docker client that does _not_ necessarily correspond 1:1 with the CLI client? And also link each Gradle task Javadoc to its underlying JSON API object? The fact that you publish blank Javadoc for all of these tasks is particularly frustrating.

**Comment by cdancy** [comment]
Lets close this issue. I don't think we want to re-document what has already been done elsewhere. We state clearly numerous times we build on top of the `docker-java` library in our documentation and if you further look at any task you can see which call we are making.
